NuRadioReco/modules/phasedarray/triggerSimulator.py

```diff
@@ -18,265 +19,284 @@
-        trigger_delays: dictionary
-            the delays for the primary channels that have caused a trigger.
-            If there is no trigger, it's an empty dictionary
-        sec_trigger_delays: dictionary
-            the delays for the secondary channels that have caused a trigger.
-            If there is no trigger or no secondary channels, it's an empty dictionary
-        """
-        station_id = station.get_id()
-
-        traces = {}
-        for channel in station.iter_channels(use_channels=triggered_channels):
-            channel_id = channel.get_id()
-            time_step = 1 / channel.get_sampling_rate()
-
-            if trigger_adc:
-
-                ADC = analogToDigitalConverter()
-
-                trace = ADC.get_digital_trace(station, det, channel,
-                                              trigger_adc=trigger_adc,
-                                              random_clock_offset=True,
-                                              adc_type='perfect_floor_comparator',
-                                              upsampling_factor=upsampling_factor,
-                                              nyquist_zone=nyquist_zone,
-                                              bandwidth_edge=bandwidth_edge)
-                time_step = 1 / det.get_channel(station_id, channel_id)['trigger_adc_sampling_frequency']
-                if upsampling_factor is not None:
-                    upsampling_factor = int(upsampling_factor)
-                    if upsampling_factor >= 2:
-                        time_step /= upsampling_factor
-                times = np.arange(len(trace), dtype=np.float) * time_step
-                times += channel.get_trace_start_time()
+            trace = traces[channel_id]
 
+            if (phased_trace is None):
+                phased_trace = np.roll(trace, subbeam_rolls[channel_id])
             else:
+                phased_trace += np.roll(trace, subbeam_rolls[channel_id])
 
-                trace = np.copy(channel.get_trace())  # get the enveloped trace
-                times = np.copy(channel.get_times())  # get the corresponding time bins
-
-            if cut_times != (None, None):
-                left_bin = np.argmin(np.abs(times - cut_times[0]))
-                right_bin = np.argmin(np.abs(times - cut_times[1]))
-                trace[0:left_bin] = 0
-                trace[right_bin:None] = 0
+            if (channel_id in sec_subbeam_rolls):
+                phased_trace += np.roll(trace, sec_subbeam_rolls[channel_id])
 
-            traces[channel_id] = trace[:]
+        # Implmentation of the ARA-like power trigger
+        window_width = int(window_time / time_step)
+        n_windows = int(len(trace) / window_width)
+        n_windows = 2 * n_windows - 1
+        squared_mean_threshold = Nant * threshold ** 2
 
-        for subbeam_rolls, sec_subbeam_rolls in zip(beam_rolls, sec_beam_rolls):
+        # Create a sliding window
```

Comment:
I think more explanation is needed to explain what the following lines are doing.